### PR TITLE
feat: support creating files with configmap in container

### LINKED ISSUE #379: Model: AppConfiguration model supports ConfigMap resource
### What would you like to be added?

Support native ConfigMap resource in AppConfiguration model. This includes:
- Definition of front-end (customer-facing) model
- Upgrade the corresponding back-end render logic
- Documentation updated to reflect the new front-end (customer-facing) model: How to use the new model, with all the fields and possible values

### Why is this needed?

This is part of the upgrade to AppConfiguration model. This enables kusion to render a Kubernetes configMap based on user input and supports the transition from the current Server model to AppConfiguration model

### Acceptance Criteria

Kusion can properly produce a valid, rendered YAML containing a configMap, based on the user input from the AppConfiguration model.
Documentations updated on Kusionstack.io.

### PR DESCRIPTION
**Pull request by liu-hm19**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:
This PR adds support for creating files with ConfigMap in workload container. 
#### Which issue(s) this PR fixes:

Fixes #379 

#### Special notes for your reviewer:
An example of creating files with ConfigMap in the container is shown below: 
image
image

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/generator/appconfiguration/generator/workload/workload_generator.go [hunk 1]
```diff
@@ -71,10 +74,16 @@ func (g *workloadGenerator) Generate(spec *models.Spec) error {
 	return nil
 }
 
-func toOrderedContainers(appContainers map[string]container.Container) ([]corev1.Container, error) {
+func toOrderedContainers(appContainers map[string]container.Container, uniqueAppName string) ([]corev1.Container, []corev1.Volume, []corev1.ConfigMap, error) {
 	// Create a slice of containers based on the app's
 	// containers.
 	var containers []corev1.Container
+
+	// Create a slice of volumes and configMaps based on the containers' files
```
Comment (SparkYuan) [inline pkg/generator/appconfiguration/generator/workload/workload_generator.go:82]: The wrap length looks too short for our big-wide monitor, you can set the wrap length at 140 :)

File: pkg/generator/appconfiguration/generator/workload/workload_generator_test.go [hunk 2]
```diff
@@ -134,17 +134,33 @@ func TestToOrderedContainers(t *testing.T) {
 			Image: "image2",
 			Env:   map[string]string{"key": "value"},
 		}
+		appContainers["container3"] = container.Container{
+			Image: "image3",
+			Files: map[string]container.FileSpec{
+				"/tmp/example.txt": {
```
Comment (SparkYuan) [inline pkg/generator/appconfiguration/generator/workload/workload_generator_test.go:140]: Add one more file to make a more complete test

File: pkg/generator/appconfiguration/generator/workload/service_generator.go [hunk 3]
```diff
@@ -76,13 +76,28 @@ func (g *workloadServiceGenerator) Generate(spec *models.Spec) error {
...
 	}
 
+	// Create ConfigMap objects based on the app's configuration.
+	for _, cm := range configMaps {
+		cm.Namespace = g.project.Name
+		if err = appconfiguration.AppendToSpec(
+			models.Kubernetes,
+			appconfiguration.KubernetesResourceID(cm.TypeMeta, cm.ObjectMeta),
+			spec,
+			&cm,
```
Comment (SparkYuan) [inline pkg/generator/appconfiguration/generator/workload/service_generator.go:95]: do not use reference to loop iterator variable. 
https://hackernoon.com/dont-make-these-5-golang-mistakes-3l3x3wcw

File: pkg/generator/appconfiguration/generator/workload/workload_generator.go (hunk 1 above)
Comment (liu-hm19) [reply pkg/generator/appconfiguration/generator/workload/workload_generator.go]: OK, I have adjusted the wrap length.

File: pkg/generator/appconfiguration/generator/workload/workload_generator_test.go (hunk 2 above)
Comment (liu-hm19) [reply pkg/generator/appconfiguration/generator/workload/workload_generator_test.go]: An additional test file has been added into the unit test case.

File: pkg/generator/appconfiguration/generator/workload/service_generator.go (hunk 3 above)
Comment (liu-hm19) [reply pkg/generator/appconfiguration/generator/workload/service_generator.go]: Oops..., sorry for not noticing this mistake.

**Review by adohe — APPROVED** [review]
overall lgtm

**Pull request merged**